test/language/expressions/function/static-init-await-binding.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) 2021 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-identifiers
+description: The restriction on `await` as a BindingIdentifier does not cross function boundaries
+info: |
+  BindingIdentifier : Identifier
+
+  [...]
+  - It is a Syntax Error if the code matched by this production is nested,
+    directly or indirectly (but not crossing function or static initialization
+    block boundaries), within a ClassStaticBlock and the StringValue of
+    Identifier is "await".
+features: [class-static-block]
+---*/
+
+class C {
+  static {
+    (function await(await) {});
```

Comment:
Can you add a new test case on function declaration?
```js
class C {
  static {
    function await(await) {};
  }
}
```
Since how the binding identifier inherits `[Await]` is different from function expressions. We can also draft some test cases based on https://github.com/tc39/proposal-class-static-block/issues/43#issuecomment-812224819

Reply:
Sounds good. Added a few more